packages/prismic/src/composables/usePrismic/index.ts

```diff
@@ -1,36 +1,65 @@
-import { ref, Ref } from '@vue/composition-api'
-import { prismic, endpoint, apiOptions } from '../../index'
-import { PrismicQuery } from '../../types'
-import { QueryOptions } from 'prismic-javascript/d.ts/ResolvedApi'
-import ApiSearchResponse from 'prismic-javascript/d.ts/ApiSearchResponse'
-import transformQuery from './transformQuery'
+import { ref, Ref } from '@vue/composition-api';
+import { prismic, endpoint, apiOptions } from '../../index';
+import { PrismicQuery, PrismicMeta } from '../../types';
+import { QueryOptions } from 'prismic-javascript/d.ts/ResolvedApi';
+import transformQuery from './transformQuery';
+import { Document } from 'prismic-javascript/d.ts/documents';
 
 interface OptionsType {
   orderings?: string;
   pageSize?: number;
   page?: number;
 }
 
-type Search = (query: PrismicQuery, options?: OptionsType) => Promise<void>
+type Search = (query: PrismicQuery | PrismicQuery[], options?: OptionsType) => Promise<void>;
 
 interface UsePrismic {
   loading: Ref<boolean>;
   error: Ref<boolean>;
-  doc: Ref<ApiSearchResponse>;
+  doc: Ref<Document | Document[]>;
+  meta: Ref<PrismicMeta | null>;
   search: Search;
+  searchFirst: Search;
 }
 
 export default function usePrismic(): UsePrismic {
   const loading = ref(false);
   const error = ref(null);
-  const doc: Ref<ApiSearchResponse> = ref({} as ApiSearchResponse);
+  const doc: Ref<Document | Document[]> = ref({});
+  const meta: Ref<PrismicMeta | null> = ref(null);
 
-  const search: Search = async (query: PrismicQuery, options: OptionsType = {}) => {
+  const search: Search = async (query: PrismicQuery | PrismicQuery[], options: OptionsType = {}) => {
+    loading.value = true;
+
+    await prismic
+      .getApi(endpoint, apiOptions)
+      .then((api) => api.query(
+        transformQuery(query),
+        options as QueryOptions
```

Comment:
if this isn't the OptionsType so maybe you should use QueryOptions instead of OptionTyme in the search definition?

Reply:
Refactored a little. As we talked, Prismic's type is not enough, that's why I've added another one